salt/states/module.py

```diff
@@ -370,6 +371,24 @@ def run(**kwargs):
 
     :return:
     """
+    # Detect if this call is using legacy or new style syntax.
+    legacy_run = True
+    keys = list(kwargs.keys())
+    if "name" in keys:
+        keys.remove("name")
+    for name in keys:
+        if not name.startswith("m_"):
+            legacy_run = False
```

Comment:
Maybe we should flip this?

```
legacy_run = any(name.startswith('m_') for name in keys)
```

Or would it be `legacy_run = all(name.startswith('m_') for name in keys)`?

Reply:
Not all the keys since you still have the `name` attribute.

Reply:
Yes, there are cases where we can have an old style call that has no `m_` prefixed arguments. I took a different approach.